Dockerfile

```diff
@@ -0,0 +1,55 @@
+FROM ubuntu:18.04
+
+LABEL maintainer="Dan Levin <dan@badpacket.in>, Martin Häcker <spamfaenger@gmx.de>"
+LABEL description="This image is intended as a developent environment for task-tracker."
+
+ENV LC_ALL=C.UTF-8 LANG=C.UTF-8  
+
+# How does docker figure out when to rerun these lines?
+# FIXME according to https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache
+# The cache for this RUN action is only invalidated if the line changes. So how do I ensure to either get updates regularly
+# or force the build process to respect some minimum or pinned versions of packages?
+RUN apt-get update \
+    && apt-get -y install \
+        python3.7 \
+        python3-pip \
+        python3-venv \
+        curl \
+        git \
+        gnupg2
+
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+    && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
+    && apt-get update \
+    && apt-get -y install yarn
+
+WORKDIR /task-tracker
+
+# JS dependencies
+COPY package.json yarn.lock /tmp/
+RUN cd /tmp && yarn install
+# TODO consider how much of this could be run on a different / dedicated container
+# Later symlink /tmp/node_packages to wherever they are needed
+# TODO How would I mount a volume to contain the download caches for yarn / pip to speed up image creation?
```

Comment:
to the best of my knowledge, you can't leverage volumes during build. you can leverage a different starting image (FROM statement) that contains the contents pre-fetched. That's what I do when I want to reduce build times.

Reply:
One other thought I could share here is that I currently split my projects into back-end and front-end containers. I tend to use the webpack dev server for its niceties (re-compliation on changes, browser reload on re-compliation, etc). In that scenario, both containers run in parallel, and their build caches are kept isolated, and that tends to also speed things up.